### DISCUSSION TITLE
[bug - macOS] - windows state is not restored when closing terminal via clicking on close button.

### DISCUSSION THREAD

**Discussion by hackmajoris** [ghostty-org/ghostty#3873] [Feature Requests, Ideas] [answered]
Reproduction steps: 
1. `window-save-state = always` in config  and reload configs.
2. layout the terminal window: add some tabs/splits
3. close the terminal via cmd + q.
4. open the Ghostty and see that the layout is restored.
5. close the Ghostty via clicking on close button.
6. open the Ghostty and see that the layout was lost

**Comment by ronit18** [comment, marked as answer]
Window state is only saved when Ghostty is fully exited (e.g., using Cmd + Q) and not when the window is closed via the red close button.

ref: https://ghostty.org/docs/config/reference#window-save-state
